### ISSUE TITLE
Implementing the C-style modulo

### ISSUE THREAD

**Issue by 0x2b3bfa0** [AlexAltea/htypes#1]
Would be great to implement the C-style modulo (the one that applies the sign of the dividend to the result) along with the Python-style modulo, because the vast majority of low-level languages use that implementation.

I mean to define a modulo operation like this:

    def mod(a, n):
        return a - int(a / n) * n

Or maybe this one, less readable you know the operator priorities in Python:

    def mod(a, n):
        return a - a // n * n

The thing is that I don't know how to implement that in your code so it becomes trivial to use. The `__mod__` operator is already used for the python `operator.mod`. I don't know if is sane to add a new method to the class or is better to replace the current one.

***

*Off-topic:* Thank you! I started a week ago to write my `int32` class for a ported code, and today I noticed that this code already existed. I'll migrate all now. Surprisingly our coding style is very similar. 😄

**Comment by AlexAltea** [comment]
I consider adhering to the C standard has priority. If people want to temporarily make Python-like divisions they can always go for `int32(int(a) % int(b))`.

In the future I intended to add extra operators/helpers for rotations (e.g. `op_ror`, `op_rol`), so I think the way to go here would be declaring a `op_mod` in [native_int.py](https://github.com/AlexAltea/ntypes/blob/master/nativetypes/native_int.py), and replacing `operator.mod` with `op_mod`. Note that due to issue #4 this won't entirely work. It turns out there's a bug with divisions I just discovered. So we would need something like:

```python
# Custom operators
def op_div(lhs, rhs):
    return int(lhs / rhs)

def op_mod(lhs, rhs):
    return lhs - lhs // rhs * rhs
```

Right now I'm busy with real life. If you can do the implementation+tests I'd really appreciate it. Otherwise, I'll take care of this next weekend.

**Comment by 0x2b3bfa0** [comment]
I'll try to implement it.

**Comment by AlexAltea** [comment]
Fixes are available on [nativetypes-1.0.3](https://pypi.python.org/pypi/nativetypes).
Thanks for your help! :-)
